src/components/ProductList/ProductList.jsx

```diff
@@ -1,88 +1,109 @@
-import React from "react";
+import React, { useState } from "react";
+import {Link} from "react-router-dom";
 import "./ProductList.css";
 import { useProductContext } from "../../contexts/ProductsListContext";
 
 const ProductList = () => {
 
-    const { product,productState:{sort, byStock, byFastDelivery, byHomeCategory, byNikon, bySony, byCanon, bySamsung, byPanasonic, byRating, bySearch, byRange } } = useProductContext()
+    const { product, productState: { sort, byStock, byFastDelivery, byHomeCategory, byNikon, bySony, byCanon, bySamsung, byPanasonic, byRating, bySearch, byRange, byCategory }, addCartState: { addCart }, addCartDispatch, wishListState: { addWishList }, wishListDispatch } = useProductContext()
```

Comment:
also the filters can be extracted into one single object here and then use it for exporting, instead of doing this